src/LayerEditor/ui/data/block_item.py

```diff
@@ -0,0 +1,142 @@
+from LayerEditor.ui.data.interpolated_node_set_item import InterpolatedNodeSetItem
+from LayerEditor.ui.data.layer_item import LayerItem
+from LayerEditor.ui.data.region_item import RegionItem
+from LayerEditor.ui.diagram_editor.graphics_items.gs_point import GsPoint
+from LayerEditor.ui.diagram_editor.graphics_items.gs_polygon import GsPolygon
+from LayerEditor.ui.diagram_editor.graphics_items.gs_segment import GsSegment
+
+from LayerEditor.ui.tools.id_map import IdMap, IdObject
+from LayerEditor.ui.tools.selection import Selection
+
+from gm_base.geometry_files.format_last import InterfaceNodeSet, NodeSet, InterpolatedNodeSet, StratumLayer, \
+    FractureLayer
+
+
+class BlockItem(IdObject):
+    """Holds common data for a block of layers"""
+    def __init__(self, regions_model):
+        super(BlockItem, self).__init__()
+        """Reference for LEData."""
+        self.regions_model = regions_model
+        """Reference to object which manages regions."""
+        self.layers = []
+        """list of layer in this block"""
+        self.layers_dict = IdMap()
+
+        self.selection = Selection()
+        """Selection is common for all layers in this block."""
+
+        self.gui_selected_layer = None
+        """Currently active layer. Region changes are made on this layer."""
+
+
+    @property
+    def decomposition(self):
+        # This is going to be problem in future when there will be more editable layers in one block
+        for layer in self.layers:
+            if isinstance(layer.top_top, InterfaceNodeSetItem):
+                return layer.top_top.decomposition
+
+    @property
+    def layer_names(self):
+        for layer in self.layers:
+            yield layer.name
+
+    def make_node_set_from_data(self, le_model, node_set: [InterfaceNodeSet, InterpolatedNodeSet]):
+        if isinstance(node_set, InterfaceNodeSet):
+            le_model.decompositions_model.decomps[node_set.nodeset_id].helper_attr_block = self
+            # I don't have any other sensible idea how to keep track of topology_id inside decomposition
+            return InterfaceNodeSetItem(le_model.decompositions_model.decomps[node_set.nodeset_id],
+                                        le_model.interfaces_model.interfaces[node_set.interface_id])
+        else:
+            interfaces = le_model.interfaces_model.interfaces
+            decomp = le_model.decompositions_model.decomps[node_set.surf_nodesets[0].nodeset_id]
+            itf = interfaces[node_set.surf_nodesets[0].interface_id]
+            itf_node_set1 = InterfaceNodeSetItem(decomp, itf)
+            decomp = le_model.decompositions_model.decomps[node_set.surf_nodesets[1].nodeset_id]
+            itf = interfaces[node_set.surf_nodesets[1].interface_id]
+            itf_node_set2 = InterfaceNodeSetItem(decomp, itf)
+
+            itf = interfaces[node_set.interface_id]
+
+            return InterpolatedNodeSetItem(itf_node_set1, itf_node_set2, itf)
+
+
```

Comment:
Originaly. Almost everything in `init_add_layer` is new.
https://github.com/GeoMop/GeoMop/blob/8ddb4310a7a18fab6e09c6563bcc32f9ab074f64/src/LayerEditor/ui/data/block_model.py#L47-L53